OpenAI_API/Chat/ChatMessage.cs

```diff
@@ -181,9 +184,12 @@ public ImageInput Image
 
 				set
 				{
-					image = value;
-					text = null;
-					Type = "image_url";
+					if (value != null)
```

Comment:
After spending a lot of time, I found that this was the root cause. Values were being overwritten, hence deserialization was breaking. I tried to think of all cases/paths, and it looks fine to me to add this check. @OkGoDoIt , if you can think of anything where this doesn't fit it, let me know.